src/bacnet/basic/object/bi.c

```diff
@@ -615,3 +861,417 @@ bool Binary_Input_Write_Property(BACNET_WRITE_PROPERTY_DATA *wp_data)
+                BI_Descr[index].Event_Time_Stamps[i];
+        }
+        /* Notify Type */
+        getevent_data->notifyType = BI_Descr[index].Notify_Type;
+        /* Event Enable */
+        bitstring_init(&getevent_data->eventEnable);
+        bitstring_set_bit(&getevent_data->eventEnable, TRANSITION_TO_OFFNORMAL,
+            (BI_Descr[index].Event_Enable & EVENT_ENABLE_TO_OFFNORMAL) ? true
+                                                                       : false);
+        bitstring_set_bit(&getevent_data->eventEnable, TRANSITION_TO_FAULT,
+            (BI_Descr[index].Event_Enable & EVENT_ENABLE_TO_FAULT) ? true
+                                                                   : false);
+        bitstring_set_bit(&getevent_data->eventEnable, TRANSITION_TO_NORMAL,
+            (BI_Descr[index].Event_Enable & EVENT_ENABLE_TO_NORMAL) ? true
+                                                                    : false);
+        /* Event Priorities */
+        Notification_Class_Get_Priorities(
+            BI_Descr[index].Notification_Class, getevent_data->eventPriorities);
+
+        return 1; /* active event */
+    } else
+        return 0; /* no active event at this index */
+}
+
+
+bool Binary_Input_Alarm_Value_Set(
+    uint32_t object_instance, BACNET_BINARY_PV value)
+{
+    unsigned index = 0;
+    bool status = false;
+
+    index = Binary_Input_Instance_To_Index(object_instance);
+    if (index < BI_Max_Index) {
+        if (Polarity[index] != POLARITY_NORMAL) {
+            if (value == BINARY_INACTIVE) {
+                value = BINARY_ACTIVE;
+            } else {
+                value = BINARY_INACTIVE;
+            }
+        }
+        BI_Descr[index].Alarm_Value = value;
+        status = true;
+    }
+
+    return status;
+}
+#endif /* (INTRINSIC_REPORTING) */
+
+void Binary_Input_Intrinsic_Reporting(uint32_t object_instance)
+{
+#if (BINARY_INPUT_INTRINSIC_REPORTING)
+    BACNET_EVENT_NOTIFICATION_DATA event_data = { 0 };
+    BACNET_CHARACTER_STRING msgText = { 0 };
+    BINARY_INPUT_DESCR *CurrentBI = NULL;
+    unsigned int object_index = 0;
+    uint8_t FromState = 0;
+    uint8_t ToState = 0;
+    BACNET_BINARY_PV Alarm_Value = BINARY_INACTIVE;
+    BACNET_BINARY_PV PresentVal  = BINARY_INACTIVE;
+    bool SendNotify = false;
+
+    object_index = Binary_Input_Instance_To_Index(object_instance);
+    if (object_index < BI_Max_Index) {
+        CurrentBI = &BI_Descr[object_index];
+    } else {
+        return;
+    }
+    /* check whether Intrinsic reporting is enabled*/
+    if (!CurrentBI->Event_Detection_Enable) {
+        return; /* limits are not configured */
+    }
+
+    if (CurrentBI->Ack_notify_data.bSendAckNotify) {
+        /* clean bSendAckNotify flag */
+        CurrentBI->Ack_notify_data.bSendAckNotify = false;
+        /* copy toState */
+        ToState = CurrentBI->Ack_notify_data.EventState;
+        PRINT("Binary-Input[%d]: Send AckNotification.\n", object_instance);
```

Comment:
Is this PRINT for debug ? do we want it in the live code

Reply:
We shall be disabling the debug printing for the production.